service/controller/resource/templateignition/create.go

```diff
@@ -25,45 +25,48 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	master, err := key.Render(cc, key.MasterTemplatePath, false)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	worker, err := key.Render(cc, key.WorkerTemplatePath, false)
-	if err != nil {
-		return microerror.Mask(err)
+	var template map[string]string
+	{
+		if cr.Spec.IsMaster {
+			template, err = key.Render(cc, key.MasterTemplatePath, false)
+		} else {
+			template, err = key.Render(cc, key.WorkerTemplatePath, false)
+		}
+		if err != nil {
+			return microerror.Mask(err)
+		}
 	}
 
-	cm := &corev1.ConfigMap{
+	s := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: key.StatusConfigMapName(cr.Spec.ClusterID),
 			Labels: map[string]string{
-				"cluster":       cr.Spec.ClusterID,
-				label.ManagedBy: project.Name(),
+				"cluster.x-k8s.io/cluster-name": cr.Spec.ClusterID,
+				label.ManagedBy:                 project.Name(),
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(&cr, cr.GroupVersionKind()),
 			},
 		},
-		Data: map[string]string{
-			"master": master["."],
-			"worker": worker["."],
+		StringData: map[string]string{
+			"value": template["."],
 		},
 	}
 
-	actualCM, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.DefaultNamespace).Update(cm)
+	actualSecret, err := r.k8sClient.K8sClient().CoreV1().Secrets(key.DefaultNamespace).Update(s)
```

Comment:
We will probably change this to place the secret in the same namespace as the CR - can do that separately though.